src/components/TopologySidebar/TopologyActionBar.tsx

```diff
@@ -0,0 +1,286 @@
+import { MdAlarmAdd, MdTableRows } from "react-icons/md";
+import { useNavigate } from "react-router-dom";
+import useUpdateComponentMutation from "../../api/query-hooks/mutations/useUpdateComponentMutation";
+import { EvidenceType } from "../../api/services/evidence";
+import { Topology } from "../../context/TopologyPageContext";
+import { AttachEvidenceDialog } from "../AttachEvidenceDialog";
+import { Button } from "../Button";
+import TopologySnapshotModal from "../TopologyCard/TopologySnapshotModal";
+
+type TopologyActionItem = {
+  /**
+   * The label to be shown in the action item
+   */
+  label: string;
+  /**
+   *
+   * The icon to be shown in the action item, this is a react component and can
+   * be rendered directly
+   *
+   * e.g. <Icon {...props} />
+   *
+   */
+  Icon: IconType;
+  /**
+   * A function that returns true if the action item should be shown
+   */
+  isShown: (
+    topology: Topology,
+    displayPoint: "TopologyCard" | "TopologySidebar"
+  ) => boolean;
+  /**
+   *
+   * A headless component that wraps the child component and provides the onClick handler
+   * and icon and text props.
+   *
+   * The onclick handler is unique to each action item, and the icon and text
+   * and hence the need to create them in the container component and pass them
+   * to child component.
+   *
+   */
+  ContainerComponent: React.FC<{
+    child: React.FC<{
+      onClick: () => void;
+      icon: React.ReactNode;
+      text: React.ReactNode;
+    }>;
+    topology: Topology;
+    onRefresh?: () => void;
+    icon: React.ReactNode;
+    text: React.ReactNode;
+    /**
+     *
+     * A callback that is called to open a modal such as the attach evidence
+     * dialog or the topology snapshot modal.
+     *
+     * This modals can not be created within the container component because
+     * they will get unmounted when the Menu dropdown is closed.
+     *
+     */
+    openModalAction?: () => void;
+  }>;
+};
+
+export const topologyActionItems: Readonly<TopologyActionItem>[] = [
+  {
+    label: "View Logs",
+    Icon: MdTableRows,
+    isShown: (topology) => !!topology.external_id,
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      icon,
+      text
+    }) {
+      const navigate = useNavigate();
+
+      return (
+        <ChildComponent
+          onClick={() => navigate(`/logs?topologyId=${topology.id}`)}
+          icon={icon}
+          text={text}
+        />
+      );
+    }
+  },
+  {
+    label: "Zoom In",
+    Icon: BiZoomIn,
+    isShown: (topology, displayPoint) => displayPoint === "TopologyCard",
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      icon,
+      text
+    }) {
+      const navigate = useNavigate();
+
+      return (
+        <ChildComponent
+          onClick={() => navigate(`/topology/${topology.id}`)}
+          icon={icon}
+          text={text}
+        />
+      );
+    }
```

Comment:
Why can't this just be onClick: {() => navigate(...)}

Reply:
The logic to determine URL to navigate to would also need to be extracted away (properties also don't work as they don't have access to the topology object needed to determine the component details), which would affect readability and ease of use when consuming the list and also adding new action items.